topology.go

```diff
@@ -25,17 +26,43 @@ type TopologyOptions struct {
 	// TODO: Filter status and types in DB Query
 	Types  []string
 	Status []string
+
+	// when set to true, only the children (except the immediate children) are returned.
+	// when set to false, the direct children & the parent itself is fetched.
+	restOfTheChildrenOnly bool
 }
 
 func (opt TopologyOptions) String() string {
 	return fmt.Sprintf("%#v", opt)
 }
 
+// selectClause returns the columns that should be selected from the topology view.
+func (opt TopologyOptions) selectClause() string {
+	if !opt.restOfTheChildrenOnly {
+		return "*"
+	}
+
+	// parents, incidents, analysis, checks columns need to fetched to create the topology tree even though they may not be essential to the UI.
+	return "name, namespace, id, is_leaf, status, status_reason, summary, topology_type, labels, team_names, agent_id, parent_id, type, parents, incidents, analysis, checks"
+}
```

Comment:
@moshloop had to add back these columns as they're essential during tree creation.
- parents
- incidents
- analysis
- checks

Reply:
I am not sure how much better this will perform - I also think we should use more sql functions / views and have this in a single db query rather than multiple trips to the db.

Generally I think there are phases:

1. Find all top level components (no parents, or direct descendants) and apply label, type, team, agent filters at this level only. 
2. For each component summarize health, incidents, analysis of its descendants 
    
```
// find all the descendants of a component, including those via relationships
find_descendants(id): returns id array
````

```
descendants =  find_descendants(id)
select components.* from components where filters
with each top level component
select name, id, parent, type, icon, component_summary.* from components where id in @descandents 
inner join component_summary
```

where component_summary is a view, that maybe becomes a materialized view in the future (which is refreshed on a debonce(check.transitioned, analysis.updated, incident.transitioned) etc..

3. Union and then post process for top-level summary in go
- strip off parent(s), agent_id in final topology returned 

wdyt @yashmehrotra  ?

Reply:
Agree with point 1 of applying filters just at top level

for point 2, are we selecting all the descendants or just summarizing them ? cause summary right now happens in go not sql, so we might want to change that